oak/client/application_client.h

```diff
@@ -59,13 +66,29 @@ class ApplicationClient {
   // Returns a grpc Channel connecting to the specified address, initialised with:
   // - Asylo channel credentials, possibly attesting the entire channel to the remote enclave
   // - Oak Policy call credentials, possibly attaching a specific Oak Policy to each call
+  //
+  // A fresh nonce is created for each newly created channel, and it is attached as an authorization
+  // bearer token to each outgoing call. Also a corresponding policy allowing access to the newly
+  // created token is attached to each outgoing call. The combined effect is that any interactions
+  // over this channel are private to the channel itself.
+  //
+  // For instance, a client may interact with a server and send some data, and the later on retrieve
+  // the data via a different call over the same channel. If the channel is closed (either
+  // intentionally or accidentally) then the previously sent data will not be accessible any more,
+  // since a new channel will be required, which will be associated with a different token.
+  //
   // See https://grpc.io/docs/guides/auth/.
   static std::shared_ptr<grpc::Channel> CreateChannel(std::string addr) {
+    NonceGenerator<kPerChannelNonceSizeBytes> nonce_generator;
+    auto channel_authorization_token_base64 = NonceToBase64(nonce_generator.NextNonce());
```

Comment:
nit: be nice to shuffle things so nonce-creation is directly next to nonce-use.

Reply:
Done.